### ISSUE TITLE
UITARS1.5 adaptation investigation for grounding task finetuning

### ISSUE THREAD

**Issue by Locke0** [ManifoldRG/software-control#171]
- How to adapt the 2U1 finetuning script for grounding task LoRA SFT? in terms of loss mostly. If you see anything regarding decoding predictions into action needs to be adapted let me know too (likely, it just spits out coordinate numbers directly in text without special decoding mechanisms), I'll be running inference on some examples in a few hours likely.
    - (probably need some digging on the action decoding and loss calculation in 2U1 for qwen2.5VL model and also take some inspiration from grounding models like GTA1 (not sure if any other grounding models like JEDI or SE-GUI, GUI^2, etc have their training code available, GTA1 trains for grounding tasks but uses GRPO instead of SFT but maybe their step wise spatial loss is a good reference. Be careful with going down rabbit holes, wouldn't want scope to creep)

- ~~What about mixture of both? 50% grounding data and 50% trajectory data as replay buffer to prevent catastrophic forgetting on UITARS1.5's temporal capability~~

Deliverables:
- [ ] minimal changes needed to adapt 2U1 finetuning script for UITARS1.5 for grounding tasks (given step action instruction, screenshot, predict action str, coordinates)

Setting tentative end date to tomorrow, so we can discuss it, but if you need more time, no worries too

**Comment by Locke0** [comment]
Crossed off question 2 because our research story focuses on visual robustness and understanding. So preventing catastrophic forgetting on trajectory capability deviates from our story

**Comment by YEESH421** [comment]
Finished investigation with notes here: https://coda.io/d/Software-Control-Models-Knowledge-Base_dzujAXxemDw/UI-TARS1-5-7B-Finetuning_su8RFivt#_luHDm0H7
